### PR TITLE
NAS-108910 / 13.0 / run fenced on single-node HA capable systems

### PR DESCRIPTION
**Pull request by yocalebo**

### CODE REVIEW

File: src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/fenced.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    a very rare scenario of data corruption when a single node system is
+    upgraded to an HA system and proper procedures aren't followed.
+    """
+    # HA licensed systems call fenced in the `event.py` plugin
+    # so don't call it here
+    if await middleware.call('failover.licensed'):
+        return
+
+    # if not HA capable hardware, then nothing to do
+    if await middleware.call('system.product_type') != 'ENTERPRISE':
```
Comment (william-gr) [inline src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/fenced.py:60]: Probably no point in doing this for R-series

Comment (yocalebo) [reply src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/fenced.py:60]: This should only return `ENTERPRISE` on HA capable hardware?

Comment (william-gr) [reply src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/fenced.py:60]: Ah, you're right, nvm me

**Comment by rick-mesta** [comment]
@william-gr / @yocalebo: should this PR target `truenas/13.0-stable` now (after subject change) ?

**Comment by yocalebo** [comment]
> @william-gr / @yocalebo: should this PR target `truenas/13.0-stable` now (after subject change) ?

Yes, I've created new PR against `truenas/13.0-stable` branch here: https://github.com/truenas/middleware/pull/6956. Asked you to approve since it's same changes just cherry-picked to 13 branch.

**Pull request closed without merging**